File: src/api/handlers/admin/channels.rs
//! 渠道管理 handlers
//!
//! 按职责切分为三个子模块：
//! - `types` — 渠道/端点 DTO 与枚举
//! - `crud`  — 渠道列表/创建/获取/更新/删除
//! - `probe` — 渠道连通性测试（含流式探测）

mod crud;
mod probe;
mod types;

pub use types::{
    CreateChannelRequest, ListChannelsQuery, PaginatedResponse, UpdateChannelRequest,
};

pub use crud::{create, delete, get, list, update};
pub use probe::test_endpoint;

File: frontend/src/api/channels.ts
import { apiClient } from './client'
import type {
  Channel,
  CreateChannelRequest,
  FetchModelsRequest,
  PaginatedResponse,
  TestEndpointRequest,
  TestEndpointResponse,
  UpdateChannelRequest,
} from './types'

export interface ChannelListParams {
  search?: string
  status?: string
  sort_by?: string
  sort_order?: string
  page?: number
  page_size?: number
}

export const channelsApi = {
  list: (params?: ChannelListParams) =>
    apiClient.get<PaginatedResponse<Channel>>('/channels', params as Record<string, string | number | undefined>),

  get: (id: string) => apiClient.get<Channel>(`/channels/${id}`),

  create: (data: CreateChannelRequest) =>
    apiClient.post<Channel>('/channels', data),

  update: (id: string, data: UpdateChannelRequest) =>
    apiClient.put<Channel>(`/channels/${id}`, data),

  delete: (id: string) => apiClient.delete<void>(`/channels/${id}`),

  fetchModels: (data: FetchModelsRequest) =>
    apiClient.post<string[]>('/models/fetch', data),

  testEndpoint: (data: TestEndpointRequest) =>
    apiClient.post<TestEndpointResponse>('/channels/test-endpoint', data),
}
